### ISSUE TITLE
SR: Settings-Security: The date field is announced as read-only but can be activated with a mouse

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#74842]
## Action Performed
1. Using Chrome+JAWS, open the URL - https://new.expensify.com/
2. Navigate using the tab key to the Account button and press Enter to activate it.
3. Navigate using the tab key to the Profile button and press Enter to activate it.
4. Navigate to the Date of Birth field and Press Enter to activate the modal.
5. Tab to focus the Date field and observe the screen reader announcement.
6. Using mouse, open the calendar icon.

Other occurrences:
1. On Send Invoice - Date

## Expected Result
The date of birth field should not be announced as read-only as it is actionable and can be accessed with mouse.

## Actual Result
The input field is announced by the screen reader as “read-only”, indicating it cannot be interacted with. However, using a mouse click activates the date picker, allowing the user to select a date.

HTML:
<input readonly="" autocorrect="on" placeholder="YYYY-MM-DD" autocapitalize="sentences" autocomplete="on" dir="auto" rows="1" spellcheck="true" virtualkeyboardpolicy="auto" aria-label="Date" role="presentation" class="css-11aywtz r-6taxm2 r-13awgt0 r-13qz1uu r-1yadl64 r-1pl0ysf r-vzf2wf r-a023e6 r-13uqrnb r-16dba41 r-rjixqe r-xd6kpl r-gy4na3 r-1qxgc49 r-105ug2t fs-mask" data-component="AnimatedComponent" data-element="Component" data-source-file="createAnimatedComponent.js" value="2025-07-01" style="--placeholderTextColor: #8B9C8F; outline-width: 0px; outline-style: none; box-shadow: none;">

## Area issue was found
Settings-Security

## Failed Checkpoint
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User Impact
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested Resolution
Remove the misleading "readonly" attribute and make it a button that opens a date picker.

Refer:
https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/

## Environment(s)
Windows Windows 11 - Chrome

https://github.com/user-attachments/assets/abfc6dde-a4ac-48f1-88e9-e50b73196585

<img width="1919" height="971" alt="Image" src="https://github.com/user-attachments/assets/12494233-1727-4cb7-826b-07d9db5bc8bb" />

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation   Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36605/community-issues/6995014
**Issue reported by:** NA
**Slack conversation:** NA

**Comment by bernhardoj** [comment]
> Date picker should be useable by keyboard. You should be able to open it using keyboard

@rushatgabhane It currently opens on press/click. Do we want to change it to opens on focus?

**Comment by bernhardoj** [comment]
We [decided to show the modal on focus](https://expensify.slack.com/archives/C01GTK53T8Q/p1773372428592399?thread_ts=1773326605.669789&cid=C01GTK53T8Q). PR is ready

cc: @rushatgabhane

**Comment by quinthar** [comment]
Approved $250 to @abzokhattab
